controllers/forms/form-model.js

```diff
@@ -0,0 +1,83 @@
+const { find, flatMap, has, get } = require('lodash');
+
+function enhanceField(field, associatedConditionalFields) {
```

Comment:
This function (and the one below it) feels a bit magic, could you add a few comments?